metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -168,52 +169,125 @@ public void publish(RdfWrapper rdf, Date recordDate, List<String> datasetIdsToRe
   private void publish(RdfWrapper rdf, Date recordDate, List<String> datasetIdsToRedirectFrom,
       boolean performRedirects) throws IndexingException {
 
-    // Convert RDF to Full Bean.
-    final RdfToFullBeanConverter fullBeanConverter = fullBeanConverterSupplier.get();
-    final FullBeanImpl fullBean = fullBeanConverter.convertRdfToFullBean(rdf);
+    final FullBeanImpl fullBean = convertRDFToFullBean(rdf);
+
+    final TriConsumer<FullBeanImpl, FullBeanImpl, Pair<Date, Date>> fullBeanPreprocessor = providePreprocessor();
+
+    final List<Pair<String, Date>> recordsForRedirection = performRedirection(rdf,
+        recordDate, datasetIdsToRedirectFrom, performRedirects);
+
+    final FullBeanImpl savedFullBean = publishToMongo(recordDate, fullBean, fullBeanPreprocessor,
+        recordsForRedirection);
+
+    publishToSolrFinal(rdf, savedFullBean);
+  }
+
+  /**
+   * Publishes an RDF only to mongo server
+   *
+   * @param rdf RDF to publish.
+   * @param recordDate The date that would represent the created/updated date of a record
+   * @throws IndexingException which can be one of:
+   * <ul>
+   * <li>{@link IndexerRelatedIndexingException} In case an error occurred during publication.</li>
+   * <li>{@link SetupRelatedIndexingException} in case an error occurred during indexing setup</li>
+   * <li>{@link RecordRelatedIndexingException} in case an error occurred related to record
+   * contents</li>
+   * </ul>
+   */
+  public void publishMongo(RdfWrapper rdf, Date recordDate) throws IndexingException {
+    final FullBeanImpl fullBean = convertRDFToFullBean(rdf);
+
+    final TriConsumer<FullBeanImpl, FullBeanImpl, Pair<Date, Date>> fullBeanPreprocessor = providePreprocessor();
 
+    publishToMongo(recordDate, fullBean, fullBeanPreprocessor, Collections.emptyList());
+  }
+
+  /**
+   * Publishes an RDF to solr server
+   *
+   * @param rdf RDF to publish.
+   * @param recordDate The date that would represent the created/updated date of a record
+   * @throws IndexingException which can be one of:
+   * <ul>
+   * <li>{@link IndexerRelatedIndexingException} In case an error occurred during publication.</li>
+   * <li>{@link SetupRelatedIndexingException} in case an error occurred during indexing setup</li>
+   * <li>{@link RecordRelatedIndexingException} in case an error occurred related to record
+   * contents</li>
+   * </ul>
+   */
+  public void publishSolr(RdfWrapper rdf, Date recordDate) throws IndexingException {
+    final FullBeanImpl fullBean = convertRDFToFullBean(rdf);
+    if (!preserveUpdateAndCreateTimesFromRdf){
+      final Date createdDate = recordDate;
```

Comment:
Is this comment supposed to explain what is happening? Then could you remove the `TODO`? it could be misleading that something is missing here
Otherwise, if this comment is not needed at all, then just remove it please

Reply:
actually this TODO is still valid, I created the followup ticket in jira. https://europeana.atlassian.net/browse/MET-5160

Reply:
it will be implemented in this feature. will create a new branch, the rebase polluted the PR.